### ISSUE TITLE
Requirments fail to pip install correctly

### ISSUE THREAD

**Issue by blaggacao** [odoo/odoo#4701]
Welcome to dependency hell:
We are here, for reference:
[requirements.txt](https://github.com/blaggacao/odoo-fig-boot2docker/blob/db2bd7906411392a9f1d561a63284718aa5661f9/DevDocker/requirements.txt)
[Dockerfile](https://github.com/blaggacao/odoo-fig-boot2docker/blob/db2bd7906411392a9f1d561a63284718aa5661f9/DevDocker/Dockerfile)

[pip install](https://github.com/blaggacao/odoo-fig-boot2docker/blob/db2bd7906411392a9f1d561a63284718aa5661f9/DevDocker/Dockerfile#L12) throws the following errors:
```
Collecting Pillow==2.5.1 (from -r requirements.txt (line 5))
  Downloading Pillow-2.5.1.zip (6.9MB)
    Single threaded build, not installing mp_compile: 1 processes
Collecting jcconv==0.2.3 (from -r requirements.txt (line 16))
  Downloading jcconv-0.2.3.tar.gz
    file jcconv.py (for module jcconv) not found
    file jcconv_test.py (for module jcconv_test) not found
Collecting lxml==3.3.5 (from -r requirements.txt (line 17))
  Downloading lxml-3.3.5.tar.gz (3.5MB)
    /usr/local/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'bugtrack_url'
      warnings.warn(msg)
    Building lxml version 3.3.5.
    Building without Cython.
    Using build configuration of libxslt 1.1.26
    Building against libxml2/libxslt in the following directory: /usr/lib
Collecting pydot==1.0.2 (from -r requirements.txt (line 24))
  Downloading pydot-1.0.2.tar.gz
    Couldn't import dot_parser, loading of dot files will not be possible.
Collecting python-ldap==2.4.15 (from -r requirements.txt (line 28))
  Downloading python-ldap-2.4.15.tar.gz (111kB)
    defines: HAVE_SASL HAVE_TLS HAVE_LIBLDAP_R
    extra_compile_args:
    extra_objects:
    include_dirs: /opt/openldap-RE24/include /usr/include/sasl /usr/include
    library_dirs: /opt/openldap-RE24/lib /usr/lib
    libs: ldap_r
    file Lib/ldap.py (for module ldap) not found
    file Lib/ldap/controls.py (for module ldap.controls) not found
    file Lib/ldap/extop.py (for module ldap.extop) not found
    file Lib/ldap/schema.py (for module ldap.schema) not found
    building '_ldap' extension
[...]
    Modules/LDAPObject.c:18:18: fatal error: sasl.h: No such file or directory

    compilation terminated.

    error: command 'gcc' failed with exit status 1

    ----------------------------------------
[...]
```
Although those are dependencies error, I put it here, as it is not the expected behaviour. Also I wanted to counter-check with you, if there are probably some hidden dependencies which are not correctly resolved by pip. Thanks!

**Comment by blaggacao** [comment]
Maybe this is a case for nix? :smile: http://nixos.org/nix/about.html

**Comment by katyukha** [comment]
Had same error when first installing on system. Look this [package.dfdebian](https://github.com/odoo/odoo/blob/8.0/setup/package.dfdebian#L19) file, that install some system packages for debian. \*-dev packages are not installable via *pip* and contain headers to build C extensions and should be provided by system's package manager.

**Comment by LeartS** [comment]
I think you need to install the `libsasl2-dev` package using the distro package manager.

**Comment by xmo-odoo** [comment]
@katyukha and @LeartS seem to have fully covered it. Thank you both.

**Comment by blaggacao** [comment]
Thanks, for the hints! I first looked at the debian/control file which only tells part of the truth.
However I ended up to go with the [wheels](https://github.com/blaggacao/odoo-fig-boot2docker/blob/master/PyEnvDocker/Dockerfile#L17) provided by xcgd people, compiling code on a production machine goes against principle anyhow. What is good for production is good for (rather) static dev environment too.. Thanks & Best!
